### ISSUE TITLE
[java] GuardLogStatement can have more detailed example

### ISSUE THREAD

**Issue by KroArtem** [pmd/pmd#3144]
**Affects PMD Version:** 
6.31

**Rule:** [GuardLogStatement](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#guardlogstatement)

**Description:**

GuardLogStatement says that you should check for logLevel or do not check for logLevel but do not create String via concatenation. It would be useful to have one more example on how to avoid `if` statement and to use [lazy logging
](https://logging.apache.org/log4j/2.x/manual/api.html) for log4j2 or [slf4j working with parameters](https://www.baeldung.com/slf4j-with-log4j2-logback#Further)
```
Whenever using a log level, one should check if the loglevel is actually enabled, or
otherwise skip the associate String creation and manipulation.
EXAMPLE:
    // Add this for performance
    if (log.isDebugEnabled() { ...
        log.debug("log something" + " and " + "concat strings");
```

**Exception Stacktrace:**

```
# Copy-paste the stack trace here
```

**Code Sample demonstrating the issue:**

```

```

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

1.  ... (e.g. if you're using maven: `mvn clean verify`)
2.  ...

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by adangel** [comment]
See also https://github.com/pmd/pmd/issues/370 and https://sourceforge.net/p/pmd/bugs/1341/

**Comment by adangel** [comment]
Proposed documentation:

> Whenever using a log level, one should check if the loglevel is actually enabled, or
> otherwise skip the associate String creation and manipulation.
>
> An alternative to checking the log level are substituting parameters, formatters or lazy logging
> with lambdas. The available alternatives depend on the actual logging framework.

The examples are now:

```java
    // Add this for performance
    if (log.isDebugEnabled() {
        log.debug("log something" + " and " + "concat strings");
    }
    // Avoid the guarding if statement with substituting parameters
    log.debug("log something {} and {}", param1, param2);
    // Avoid the guarding if statement with formatters
    log.debug("log something %s and %s", param1, param2);
    // Avoid the guarding if statement with lazy logging and lambdas
    log.debug("log something expensive: {}", () -> calculateExpensiveLoggingText());
```

@KroArtem What do you think?

**Comment by KroArtem** [comment]
@adangel , yep, I think this is the most appropriate clarification. Since different logging frameworks have different options, enumerating all possible ways to avoid guarding statement is a good thing.

**Comment by durimkryeziu** [comment]
I have a log statement as in the example docs like: `log.debug("log something {} and {}", param1, param2);` and PMD still complains? Why so?
